src/openforms/submissions/tests/test_start_submission.py

```diff
@@ -231,3 +231,19 @@ def test_start_submission_with_prefill(self, mock_logevent):
         self.assertEqual(
             SubmissionValueVariableSources.prefill, prefilled_variable.source
         )
+
+    def test_start_submission_with_initial_data_reference(self):
+        body = {
+            "form": f"http://testserver.com{self.form_url}",
+            "formUrl": "http://testserver.com/my-form",
+            "initialDataReference": "of-or-3452fre3",
+        }
+
+        response = self.client.post(self.endpoint, body)
+
+        submission = Submission.objects.get()
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(
+            submission.initial_data_reference, body["initialDataReference"]
+        )
```

Comment:
in tests, please make assertions with the literal values instead of referring to variables in tests, that makes it clear what is being compared to what.